core/http/client/src/main/java/org/eclipse/rdf4j/http/client/SparqlSession.java

```diff
@@ -988,10 +994,12 @@ private HttpResponse sendBooleanQueryViaHttp(HttpUriRequest method, Set<QueryRes
 					acceptParam += ";q=0." + qValue;
 				}
 
-				method.addHeader(ACCEPT_PARAM_NAME, acceptParam);
+				acceptValues.add(acceptParam);
 			}
 		}
 
+		method.addHeader(ACCEPT_PARAM_NAME, Joiner.on(", ").join(acceptValues));
```

Comment:
Joiner is thread safe, we should probably extract that as a static variable in the class. Same for the joiner above.